source/linker.cpp

```diff
@@ -0,0 +1,1098 @@
+      break;
+    case spv::Op::OpGroupFAdd:
+    case spv::Op::OpGroupFMin:
+    case spv::Op::OpGroupFMax:
+    case spv::Op::OpSubgroupBallotKHR:
+      TransformConsecutiveIDs(1u, insn_word_count);
+      break;
+    case spv::Op::OpSubgroupFirstInvocationKHR:
+    case spv::Op::OpEnqueueMarker:
+    case spv::Op::OpEnqueueKernel:
+    case spv::Op::OpGetKernelNDrangeSubGroupCount:
+    case spv::Op::OpGetKernelNDrangeMaxSubGroupSize:
+    case spv::Op::OpGetKernelWorkGroupSize:
+    case spv::Op::OpGetKernelPreferredWorkGroupSizeMultiple:
+      TransformConsecutiveIDs(1u, insn_word_count);
+      RegisterVariable(i + 1u, i + 2u);
+      break;
+    case spv::Op::OpRetainEvent:
+    case spv::Op::OpReleaseEvent:
+    case spv::Op::OpCreateUserEvent:
+    case spv::Op::OpIsValidEvent:
+    case spv::Op::OpSetUserEventStatus:
+    case spv::Op::OpCaptureEventProfilingInfo:
+    case spv::Op::OpGetDefaultQueue:
+    case spv::Op::OpBuildNDRange:
+      TransformConsecutiveIDs(1u, insn_word_count);
+      break;
+    case spv::Op::OpReadPipe:
+    case spv::Op::OpWritePipe:
+    case spv::Op::OpReservedReadPipe:
+    case spv::Op::OpReservedWritePipe:
+    case spv::Op::OpGetNumPipePackets:
+    case spv::Op::OpGetMaxPipePackets:
+      TransformConsecutiveIDs(1u, insn_word_count);
+      RegisterVariable(i + 1u, i + 2u);
+      break;
+    case spv::Op::OpReserveReadPipePackets:
+    case spv::Op::OpReserveWritePipePackets:
+    case spv::Op::OpCommitReadPipe:
+    case spv::Op::OpCommitWritePipe:
+    case spv::Op::OpIsValidReserveId:
+    case spv::Op::OpGroupReserveReadPipePackets:
+    case spv::Op::OpGroupReserveWritePipePackets:
+    case spv::Op::OpGroupCommitReadPipe:
+    case spv::Op::OpGroupCommitWritePipe:
+      TransformConsecutiveIDs(1u, insn_word_count);
+      break;
+    case spv::Op::Max:
+      return libspirv::DiagnosticStream(position, context.consumer,
+                                        SPV_ERROR_INVALID_DATA)
+             << "Invalid opcode: OpMax.";
+    default:
+      return libspirv::DiagnosticStream(position, context.consumer,
+                                        SPV_ERROR_INVALID_DATA)
+             << "Unsupported opcode: " << static_cast<uint32_t>(opcode) << ".";
+    }
+
+    position.index = i += insn_word_count;
+  }
+
+  return SPV_SUCCESS;
+}
+
+static const char* AddressingModelStr(spv::AddressingModel model) {
+  switch (model) {
+  case spv::AddressingModel::Logical:    return "Logical";
+  case spv::AddressingModel::Physical32: return "Physical32";
+  case spv::AddressingModel::Physical64: return "Physical64";
+  case spv::AddressingModel::Max:        return "Unsupported";
+  }
+}
+
+static const char* MemoryModelStr(spv::MemoryModel model) {
+  switch (model) {
+  case spv::MemoryModel::Simple:  return "Simple";
+  case spv::MemoryModel::GLSL450: return "GLSL450";
+  case spv::MemoryModel::OpenCL:  return "OpenCL";
+  case spv::MemoryModel::Max:     return "Unsupported";
+  }
```

Comment:
Similar to above, do a lookupOperand to get the name, but this time with  SPV_OPERAND_TYPE_MEMORY_MODEL